### PR TITLE
[CHAT-54] "Authentification and authorization". Add auth0, add custom authentificaton jwt

### PR DESCRIPTION
**Pull request by Raigester**
Add authentification auth0.com, add custom jwt token verifier

### CODE REVIEW

File: voicenger/settings/base.py [hunk 1]
```diff
@@ -149,18 +152,33 @@
...
-    'AUTH_HEADER_TYPES': ('Bearer',),
-    'AUTH_TOKEN_CLASSES': ('rest_framework_simplejwt.tokens.AccessToken',),
-    'TOKEN_TYPE_CLAIM': 'token_type',
-}
+# SIMPLE_JWT = {
+#     'ACCESS_TOKEN_LIFETIME': timedelta(minutes=5),
+#     'REFRESH_TOKEN_LIFETIME': timedelta(days=1),
+#     'ROTATE_REFRESH_TOKENS': False,
+#     'BLACKLIST_AFTER_ROTATION': True,
+#     'ALGORITHM': 'HS256',
```
Comment (VladPrime11) [inline voicenger/settings/base.py:160]: I thought you said you had to use RS256.

File: voicenger/settings/base.py [hunk 2]
```diff
@@ -149,18 +152,33 @@
...
+#     'AUTH_HEADER_TYPES': ('Bearer',),
+#     'AUTH_TOKEN_CLASSES': ('rest_framework_simplejwt.tokens.AccessToken',),
+#     'TOKEN_TYPE_CLAIM': 'token_type',
+# }
+
+# SIMPLE_JWT = {
+#     'ALGORITHM': 'RS256',
+#     'VERIFYING_KEY': 'KbBNx2MGrxhkmzqn0LMkp6W5oXomLtgc',
+#     'AUDIENCE': 'https://dev-m4yp61l880yrkjnn.us.auth0.com/api/v2/',
+#     'ISSUER': 'https://dev-m4yp61l880yrkjnn.us.auth0.com/',
```
Comment (VladPrime11) [inline voicenger/settings/base.py:172]: Is that how it's supposed to be in the public view?

**Review by VladPrime11 — COMMENTED** [review]
There are a couple of clarifying questions. Other than that, everything is fine.

File: voicenger/settings/base.py (hunk 2 above)
Comment (Raigester) [reply voicenger/settings/base.py:172]: There is no secret data in this line, these are public keys that are already visible during authorization, but I agree it is better to remove them.

Comment (VladPrime11) [reply voicenger/settings/base.py:172]: Perhaps it makes sense to put them in an .env file?

**Review by VladPrime11 — COMMENTED** [review]
All done

**Review by VladPrime11 — APPROVED** [review]

**Pull request merged**